### PR TITLE
Introduce optFlatMap and OptionalTypes documentation

### PR DESCRIPTION
**Pull request by TristonianJones**
The `optFlatMap` macro allows for the use of optional syntax within mapping
expressions which are applied if a value exists. 

This feature exists at `OptionalTypesVersion(1)` and above.

### CODE REVIEW

File: ext/strings.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ const (
 //	'hello'.charAt(5)  // return ''
 //	'hello'.charAt(-1) // error
 //
-// # Format
+// # Strings.Format
```
Comment (l46kok) [inline ext/strings.go]: nit: Does this need to be prefixed with a namespace? It's a receiver call on an existing string

File: cel/library.go [hunk 2]
```diff
@@ -132,29 +134,182 @@ func (stdLibrary) ProgramOptions() []ProgramOption {
...
+//	map[?key]
+//
+// ## Optional Field Setting
+//
+// When creating map or message literals, if a field may be optionally set
+// based on its presence, then placing a `?` before the field name or key
+// will ensure the type on the right-hand side must be optional(T) where T
+// is the type of the field or key-value.
+//
+//		// returns a map with the key expression set only if the subfield is
```
Comment (l46kok) [inline cel/library.go]: nit: odd indentation

File: ext/strings.go (hunk 1 above)
Comment (TristonianJones) [reply ext/strings.go]: My mistake. Undone.

**Comment by TristonianJones** [comment]
PTAL

**Review by l46kok — APPROVED** [review]

**Pull request merged**